datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -858,4 +911,26 @@ mod test {
 ]"###;
         assert_eq!(expected, formatted_fields_with_datatype);
     }
+
+    #[test]
+    fn filter_schema_changed() -> Result<()> {
+        let table_scan = test_table_scan()?;
+
+        let plan = LogicalPlanBuilder::from(table_scan)
+            .filter(binary_expr(
+                binary_expr(lit(1), Operator::Gt, col("a")),
+                Operator::And,
+                binary_expr(lit(1), Operator::Gt, col("a")),
```

Comment:
You might be able to write this in a more "fluent" style if you wanted

```suggestion
            .filter(lit(1).gt(col("a"))).and(
                      lit(1).gt(col("a"))),
```